docs/up-to-date-check.md

```diff
@@ -0,0 +1,84 @@
+# Up-to-date Check
```

Comment:
The normal convention here is that only docs that we expect to point end-users to go in this directory, and docs about how to work in the repo, the implementation, etc go in the `docs/repo` directory.  This seems like more of the latter.

Reply:
Makes sense. I'll move it in a subsequent PR as #4023 follows on from this PR and modifies the file.